api/envoy/extensions/filters/http/transform/v3/transform.proto

```diff
@@ -0,0 +1,83 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.transform.v3;
+
+import "envoy/config/common/mutation_rules/v3/mutation_rules.proto";
+import "envoy/config/core/v3/substitution_format_string.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.transform.v3";
+option java_outer_classname = "TransformProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/transform/v3;transformv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Transform filter configuration]
+// Transform filter :ref:`configuration overview <config_http_filters_transform>` to perform
+// HTTP header and body transformations.
+// [#extension: envoy.filters.http.transform]
+
+// Configuration for the transform filter. The request and response transformations are
+// independent and could be configured separately.
+// Only JSON body transformation is supported for now.
+message TransformConfig {
+  // Configuration for transforming request. If set then the request will be buffered
+  // until the entire body is received and transformed for JSON request bodies.
```

Comment:
Do you think this needs to also address HTTP trailers case?
(I'm not saying to add it now, but there may need to be a config knob in the future that decides what needs to be buffered before sending it down/up the stream. I believe this may be in the "Transformation" message).